pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -43,37 +43,42 @@ const (
 	maxDurationBeforeRetry time.Duration = 2 * time.Minute
 )
 
-// GoRoutineMap defines the supported set of operations.
-type GoRoutineMap interface {
-	// Run adds operationName to the list of running operations and spawns a new
-	// go routine to execute the operation. If an operation with the same name
-	// already exists, an error is returned. Once the operation is complete, the
-	// go routine is terminated and the operationName is removed from the list
-	// of executing operations allowing a new operation to be started with the
-	// same name without error.
-	Run(operationName string, operationFunc func() error) error
+// GoroutineMap defines the supported set of operations.
+type GoroutineMap interface {
+	// Run adds operationName/operationSubName to the list of running operations
+	// and spawns a new go routine to execute operationFunc.
+	// If an operation with the same operationName and no sub name already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// If an operation with the same operationName and operationSubName already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// Once the operation is complete, the go routine is terminated and the
+	// operationName/operationSubName is removed from the list of executing
+	// operations allowing a new operation to be started with the same name
+	// without error.
+	Run(operationName, operationSubName string, operationFunc func() error) error
```

Comment:
Agree on splitting.  Alternatively, you could reorder the function like:

```
Run(fn func() error, targets ...string)
```

and have identical targets block (you might have to join the targets to do that or create a special map key, so it's not hugely important to me).  Either way making clear that target has to be unique.